### PR TITLE
[Merged by Bors] - feat(LinearAlgebra/Dual): generalize results from Free to Projective

### PR DESCRIPTION
**Pull request by alreadydone**
Also show that:

+ A finite projective module is also finitely presented (generalized from Free)

+ A direct summand of a reflexive module is reflexive. In particular, a finite projective module is reflexive.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/LinearAlgebra/Dual.lean [hunk 1]
```diff
@@ -630,6 +639,21 @@ lemma dualMap_dualMap_eq_iff_of_injective
...
+/-- A direct summand of a reflexive module is reflexive. -/
+lemma IsReflexive.of_split (i : N →ₗ[R] M) (s : M →ₗ[R] N) (H : s ∘ₗ i = .id) :
+    IsReflexive R N where
+  bijective_dual_eval' :=
+    ⟨.of_comp (f := i.dualMap.dualMap) <|
+      (bijective_dual_eval R M).1.comp (injective_of_comp_eq_id i _ H),
+    .of_comp (g := s) <| (surjective_of_comp_eq_id i.dualMap.dualMap s.dualMap.dualMap <|
+      congr_arg (dualMap ∘ dualMap) H).comp (bijective_dual_eval R M).2⟩
+
+instance [Module.Finite R N] [Projective R N] : IsReflexive R N :=
```
Comment (erdOne) [inline Mathlib/LinearAlgebra/Dual.lean:652]:
```suggestion
instance (priority := 900) [Module.Finite R N] [Projective R N] : IsReflexive R N :=
```

Comment (alreadydone) [reply Mathlib/LinearAlgebra/Dual.lean:652]: The existing instance [Module.IsReflexive.of_finite_of_free](https://leanprover-community.github.io/mathlib4_docs/Mathlib/LinearAlgebra/Dual.html#Module.IsReflexive.of_finite_of_free) doesn't specify a priority. Should it?

Comment (erdOne) [reply Mathlib/LinearAlgebra/Dual.lean:652]: I think they should.

Comment (erdOne) [reply Mathlib/LinearAlgebra/Dual.lean:652]: I think the rule of thumb is instances whose result type is `Class X Y Z` where `X Y Z` are free variables (without extra structures) should have lowered priority.

**Comment by erdOne** [comment]
Thanks!
maintainer merge

**Review by jcommelin — COMMENTED** [review]
bors d=erdOne

File: Mathlib/LinearAlgebra/Dual.lean (hunk 1 above)
Comment (alreadydone) [reply Mathlib/LinearAlgebra/Dual.lean:652]: Okay, I get it.

**Comment by erdOne** [comment]
bors merge

**Pull request closed without merging**